services/motion/builtin/builtin.go

```diff
@@ -231,52 +233,183 @@ func (ms *builtIn) MoveOnGlobe(
 	kinematicsOptions.GoalRadiusMM = math.Min(motionCfg.PlanDeviationM*1000, 3000)
 	kinematicsOptions.HeadingThresholdDegrees = 8
 
-	plan, kb, err := ms.planMoveOnGlobe(
-		ctx,
-		componentName,
-		destination,
-		movementSensorName,
-		obstacles,
-		kinematicsOptions,
-		extra,
-	)
+	positionPollingPeriod := time.Duration(1000/motionCfg.PositionPollingFreqHz) * time.Millisecond
+	obstaclePollingPeriod := time.Duration(1000/motionCfg.ObstaclePollingFreqHz) * time.Millisecond
+
+	movementSensor, ok := ms.movementSensors[movementSensorName]
+	if !ok {
+		return false, resource.DependencyNotFoundError(movementSensorName)
+	}
+
+	successChan := make(chan bool)
+	defer close(successChan)
+
+	// TODO: I know this has to change
+	type planAndBase struct {
+		base kinematicbase.KinematicBase
+		plan motionplan.Plan
+	}
+	planChan := make(chan planAndBase)
+	defer close(planChan)
+
+	replanChan := make(chan bool, 1)
+	defer close(replanChan)
+	replanChan <- true
+
+	errChan := make(chan error)
+	defer close(errChan)
+
+	cancelCtx, cancelFn := context.WithCancel(context.Background())
+	ms.cancelFn = cancelFn
+	defer ms.cancelFn()
+
+	var backgroundWorkers sync.WaitGroup
+	defer backgroundWorkers.Wait()
+
+	// helper function to manage polling functions
+	startPolling := func(ctx context.Context, period time.Duration, fn func(context.Context) error) {
+		backgroundWorkers.Add(1)
+		goutils.ManagedGo(func() {
+			ticker := time.NewTicker(period)
+			defer ticker.Stop()
+			for {
+				select {
+				case <-ctx.Done():
+					errChan <- ctx.Err()
+					return
+				case <-ticker.C:
+					if err := fn(ctx); err != nil {
+						errChan <- err
+					}
+					return
+				}
+			}
+		}, backgroundWorkers.Done)
+	}
+
+	// start goroutine to execute plan
+	backgroundWorkers.Add(1)
+	goutils.ManagedGo(func() {
+		for {
+			select {
+			case <-ctx.Done():
+				errChan <- ctx.Err()
+				return
+			case planBase := <-planChan:
```

Comment:
[organizational question] why is this goroutine above the replan goroutine?
since we initialize the `planChan` to be empty I think the first couple of iterations of this for loop would do nothing..
could you more this to be below the replan goroutine? what are your thoughts?